metrics/metrics.go

```diff
@@ -0,0 +1,150 @@
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/jtblin/kube2iam/version"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const namespace = "kube2iam"
+
+var (
+	// IamRequestCount counts number of outbound requests made to AWS IAM service.
+	IamRequestCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: "iam",
+			Name:      "requests_total",
+			Help:      "Number of outbound requests made to AWS IAM service.",
+		},
+
+		[]string{
+			// The HTTP status code AWS returned
+			"code",
+			// The arn of the IAM role being requested
+			"role_arn",
+		},
+	)
+
+	// IamRequestSec tracks timing of IAM requests.
+	IamRequestSec = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: "iam",
+			Name:      "request_duration_seconds",
+			Help:      "Time taken to complete IAM request in seconds.",
+
+			Buckets: prometheus.ExponentialBuckets(0.001, 2, 14),
+		},
+		[]string{
+			// The HTTP status code AWS returned
+			"code",
+			// The arn of the IAM role being requested
+			"role_arn",
+		},
+	)
+
+	// HTTPRequestCount counts number of HTTP requests served split by handler.
+	HTTPRequestCount = prometheus.NewCounterVec(
```

Comment:
Request count metric will be exposed by the `HTTPRequestSec` histogram, so this metric is a bit redundant.

Reply:
Good point, didn't realise this when I made the metrics.